src/org/graphstream/algorithm/LongestPath.java

```diff
@@ -0,0 +1,144 @@
+        longestPath = new ArrayList<>();
+    }
+
+    public void compute() {
+        initializeAlgorithm();
+        TopologicalSort aTopoSortAlgorithm = new TopologicalSort(TopologicalSort.SortAlgorithm.DEPTH_FIRST);
+        aTopoSortAlgorithm.init(graph);
+        aTopoSortAlgorithm.compute();
+        Node[] aSortedArray = aTopoSortAlgorithm.getSortedArray();
+        if (weighted) {
+            fillDistanceMapWeighted(aSortedArray);
+        } else {
+            fillDistanceMapUnweighted(aSortedArray);
+        }
+        Map.Entry<Node, Double> maxEntry = getMaxEntryOfMap();
+        longestPathNode = maxEntry;
+        longestPath.add(maxEntry.getKey());
+        getMaxNeigbourgh(maxEntry.getKey());
+        Collections.reverse(longestPath);
+    }
+
+    private void fillDistanceMapWeighted(Node[] theSortedArray) {
+        for (Node aNode : theSortedArray) {
+            for (Edge anEdge : aNode.getEachEnteringEdge()) {
+                Node aSourceNode = anEdge.getSourceNode();
+                Node aTargetNode = anEdge.getTargetNode();
+                double aWeight = anEdge.getNumber("weight");
+                Double aMaxDistance = Math.max(distanceMap.get(aTargetNode), distanceMap.get(aSourceNode) + aWeight);
+                distanceMap.put(aTargetNode, aMaxDistance);
+            }
+        }
+    }
+
+    private void fillDistanceMapUnweighted(Node[] theSortedArray) {
+        for (Node aNode : theSortedArray) {
+            for (Edge anEdge : aNode.getEachEnteringEdge()) {
+                Node aSourceNode = anEdge.getSourceNode();
+                Node aTargetNode = anEdge.getTargetNode();
+                Double aMaxDistance = Math.max(distanceMap.get(aTargetNode), distanceMap.get(aSourceNode)) + 1;
+                distanceMap.put(aTargetNode, aMaxDistance);
+            }
+        }
+    }
+
+    private Map.Entry<Node, Double> getMaxEntryOfMap() {
+        Map.Entry<Node, Double> maxEntry = null;
+        for (Map.Entry<Node, Double> entry : distanceMap.entrySet()) {
+            if (maxEntry == null || entry.getValue().compareTo(maxEntry.getValue()) > 0) {
+                maxEntry = entry;
+            }
+        }
+        return maxEntry;
+    }
+
+    private void getMaxNeigbourgh(Node theNode) {
+        Node aMaxNode = null;
+        double aMaxDistance = 0.0;
+        for (Edge anEdge : theNode.getEachEnteringEdge()) {
+            Node aSourceNode = anEdge.getSourceNode();
+            if (distanceMap.get(aSourceNode) >= aMaxDistance) {
+                aMaxDistance = distanceMap.get(aSourceNode);
+                aMaxNode = aSourceNode;
+            }
+        }
+        if (aMaxNode != null) {
+            longestPath.add(aMaxNode);
+            getMaxNeigbourgh(aMaxNode);
+        }
+
+    }
+
+    private void initializeAlgorithm() {
+        for (Node aNode : graph.getEachNode()) {
+            for (Edge anEdge : aNode.getEachEdge()) {
+                double aWeight = anEdge.getNumber("weight");
```

Comment:
Could you provide something like `weightAttribute` (with getter and setter) and use it instead of hardcoded "weight". Like this, the algorithm will still be usable if the weights are stored in a different attribute.

Reply:
Not at all, sorry if I was not clear, I meant methods *in your class*. Something like this:

```java
Graph g = new SingleGraph("test");
g.read("test.dgs"); // Let's say in this file weights are stored in an attribute "foo"
LongestPath lp = new LongestPath();
lp.setWeightAttribute("foo");
lp.init(g);
lp.compute();
```

Reply:
example here: https://github.com/graphstream/gs-algo/blob/ef1884226869c237500823e40218870e0a830c12/src/org/graphstream/algorithm/Kruskal.java#L205